src/lib.rs

```diff
@@ -1001,3 +874,42 @@ fn full_fence() {
         atomic::fence(Ordering::SeqCst);
     }
 }
+
+/// Indicate that we're using spin-based contention and that we should yield the CPU.
+#[inline]
+fn yield_now() {
+    #[cfg(feature = "std")]
+    std::thread::yield_now();
+
+    #[cfg(not(feature = "std"))]
+    #[allow(deprecated)]
+    sync::atomic::spin_loop_hint();
+}
+
+impl Event {
+    /// Locks the event.
+    ///
+    /// This is useful for simulating contention, but otherwise serves no other purpose for users.
+    /// It is used only in testing.
```

Comment:
"it is used only in testing" then it should be behind a feature flag.
e.g. `#[cfg(any(feature = "test", test))]`

Reply:
Adding this fails with:

```
error[E0599]: no method named `__lock_event` found for struct `Event` in the current scope
  --> tests/queue.rs:24:22
   |
24 |     let lock = event.__lock_event();
   |                      ^^^^^^^^^^^^ method not found in `Event`

For more information about this error, try `rustc --explain E0599`.
```

I've added a `__test` feature that should circumvent this issue that is enabled in testing.